pkg/config/config.go

```diff
@@ -124,15 +124,31 @@ type TestConfig struct {
 	NumServices         int       `validate:"gte=0"`
 	NumPods             int       `validate:"gte=0"`
 	HostNetwork         bool
-	TestNamespace       string `default:"testns"`
-	Iterations          int    `validate:"gte=0"`
-	Duration            int    `default:"60"`
+	TestNamespace       string      `default:"testns"`
+	Iterations          int         `validate:"gte=0"`
+	Duration            int         `default:"60"`
+	DNSPerf             *DNSConfig  `validate:"required_if=TestKind dnsperf"`
+	Perf                *PerfConfig `validate:"required_if=TestType thruput-latency,required_if=TestType iperf"`
 	CalicoNodeCPULimit  string
-	DNSPerfNumDomains   int         `validate:"gte=0"`
-	DNSPerfMode         DNSPerfMode `validate:"omitempty,oneof=Inline NoDelay DelayDeniedPacket DelayDNSResponse"`
 	LeaveStandingConfig bool
 }
 
+// PerfConfig details which tests to run in thruput-latency and iperf tests.
+type PerfConfig struct {
+	Direct          bool   // Whether to do a direct pod-pod test
+	Service         bool   // Whether to do a pod-service-pod test
+	External        bool   // Whether to test from this container to the external IP for an external-service-pod test
+	ControlPort     int    // The port to use for the control connection in tests.  Used by qperf tests.
+	TestPort        int    // The port to use for the test connection in tests.  Used by qperf and iperf tests
```

Comment:
are these only used for external-service-pod tests

Reply:
No, these are used for all qperf and iperf tests.  They do get defaulted if not specified however.